packages/react/src/contents/hooks/useCommercePages.ts

```diff
@@ -1,91 +1,87 @@
- * content will be automatically called so there is no need to refetch.
- *
- * @param slug      - Slug of the commerce page to fetch for data.
- * @param params    - The target parameters that a commerce page is configured.
- * @param pageIndex - Number of the page to get, starting at 1. The default is 1.
- * @param pageSize  - Size of each page, as a number between 1 and 180. The default is 60.
- * @param strategy  - Strategy to get best ranking commerce page.
- *
- * @returns - Returns actions and selectors for commerce page data.
- */
-const useCommercePages = (
-  slug: string,
-  params: CommercePagesParams,
-  pageIndex?: number,
-  pageSize?: number,
-  strategy = 'default',
+const useCommercePages = <T = ComponentType[]>(
+  options: UseCommercePageOptions,
 ) => {
-  const queryCommerce = {
-    type: params.type,
-    id: params?.id,
-    gender: params?.gender,
-    brand: params?.brand,
-    category: params?.category,
-    priceType: params?.priceType,
-    sku: params?.sku,
-    pageSize,
-    pageIndex,
-  };
-  const queryContent = {
-    codes: slug,
-    contentTypeCode: 'commerce_pages',
-    pageIndex,
-    pageSize,
-  };
+  const {
+    enableAutoFetch = true,
+    fetchConfig,
+    strategy = Strategy.Default,
+    ...fetchQuery
+  } = options;
+
+  const query = useMemo(
+    () => ({
+      contentTypeCode: ContentTypeCode.CommercePages,
+      ...fetchQuery,
+    }),
+    [fetchQuery],
+  );
   const error = useSelector((state: StoreState) =>
-    getContentError(state, queryContent),
+    getContentError(state, query),
   );
   const isLoading = useSelector((state: StoreState) =>
-    isContentLoading(state, queryContent),
+    isContentLoading(state, query),
+  );
+  const isFetched = useSelector((state: StoreState) =>
+    isContentFetched(state, query),
   );
-  const commercePage = useSelector((state: StoreState) =>
-    getContents(state, queryContent),
+  const entries = useSelector((state: StoreState) =>
+    getContents<T>(state, query),
+  );
+  const contentGroup = useSelector((state: StoreState) =>
+    getContentByQuery(state, query),
+  );
+  const fetchCommercePagesAction = useAction(fetchCommercePages);
+
+  const fetch = useCallback(
+    () => fetchCommercePagesAction(fetchQuery, strategy, fetchConfig),
+    [fetchCommercePagesAction, fetchQuery, strategy, fetchConfig],
+  );
+
+  const pagination = useMemo(
+    () => ({
+      pageIndex: contentGroup?.number,
+      totalPages: contentGroup?.totalPages,
+      totalItems: contentGroup?.totalItems,
+    }),
+    [contentGroup],
   );
-  const action = useAction(fetchCommercePagesAction);
-  const fetchCommercePages = () => action(queryCommerce, slug, strategy);
-  const resetContents = useAction(resetContentsAction);
 
   useEffect(() => {
-    !commercePage && fetchCommercePages();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [commercePage, slug]);
+    if (!isLoading && !error && !isFetched && enableAutoFetch) {
```

Comment:
You only need this:

```suggestion
    if (!isLoading && !isFetched && enableAutoFetch) {
```

`isFetched` will always be false when there is no error or no data but it will also be false when `isLoading` is true, so we need to check for `isLoading` as well to exclude that case.